drivers/ads101x/ads101x.c

```diff
@@ -216,3 +216,70 @@ int ads101x_set_alert_parameters(const ads101x_alert_t *dev,
 
     return ADS101X_OK;
 }
+
+#if MODULE_ADC_NG
+
+
+int ads101x_adc_ng_init(void* handle, uint8_t chn, uint8_t res, uint8_t ref)
+{
+    ads101x_t *dev = (ads101x_t *)handle;
+    DEBUG("%s: dev %p, chn %u\n", __func__, dev, chn);
+
+    assert(dev != NULL);
+    assert(chn < dev->ch_numof);
+    assert(res == dev->res);
+
+    uint8_t mux_gain = 0;
+
+    if (dev->params.device == ADS101X_DEV_ADS1013 ||
+        dev->params.device == ADS101X_DEV_ADS1113) {
+        assert(ref == 0);
+        mux_gain = ADS101X_PGA_FSR_2V048;
+    }
+    else {
+        mux_gain = (6 - ref) << 1;
+    }
+
+    switch (chn) {
+        case 0: mux_gain |= ADS101X_AIN0_SINGM; break;
+        case 1: mux_gain |= ADS101X_AIN1_SINGM; break;
+        case 2: mux_gain |= ADS101X_AIN2_SINGM; break;
+        case 3: mux_gain |= ADS101X_AIN3_SINGM; break;
+        default: return -1;
+    }
+
+    /* set the channel for next conversion */
+    if (ads101x_set_mux_gain(dev, mux_gain) != ADS101X_OK) {
+        return -1;
+    }
+
+    return 0;
+}
+
+int ads101x_adc_ng_single(void *handle, int32_t *dest)
+{
+    ads101x_t *dev = (ads101x_t *)handle;
+    DEBUG("%s: dev %p, dest %p\n", __func__, dev, dest);
+
+    assert(dev != NULL);
+
+    int16_t raw;
+    /* execute one conversion for the selected channel */
+    if (ads101x_read_raw(dev, &raw) != 0) {
+        return -1;
```

Comment:
```suggestion
        return -EIO;
```

Maybe that return value would be more informative during debugging?